consensus/bor/heimdall/client.go

```diff
@@ -299,3 +352,37 @@ func (h *HeimdallClient) Close() {
 	close(h.closeCh)
 	h.client.CloseIdleConnections()
 }
+
+func sendMetrics(request *Request, failed bool) {
+	if failed {
+		switch request.reqType {
+		case StateSyncRequest:
+			stateSyncInvalidRequestMeter.Mark(1)
+			stateSyncRequestTimer.Update(time.Since(request.start))
+		case SpanRequest:
+			spanInvalidRequestMeter.Mark(1)
+			spanRequestTimer.Update(time.Since(request.start))
+		case CheckpointRequest:
+			checkpointInvalidRequestMeter.Mark(1)
+			checkpointRequestTimer.Update(time.Since(request.start))
+		case CheckpointCountRequest:
+			checkpointCountInvalidRequestMeter.Mark(1)
+			checkpointCountRequestTimer.Update(time.Since(request.start))
+		}
+	} else {
+		switch request.reqType {
+		case StateSyncRequest:
+			stateSyncValidRequestMeter.Mark(1)
```

Comment:
could we have a map[ReuestKey]map[isFailed]struct{Meter, Timer} ? I believe it'll remove most of code duplication.

Reply:
I see, adding it.

Reply:
Will add it once I am able to figure out the reasons for the metrics not showing up on datadog. With the independent metrics it's easier to debug.